File: selenium/src/test/java/com/qacommercelab/pages/ContactPage.java
package com.qacommercelab.pages;

import com.qacommercelab.utils.ConfigReader;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.Select;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;

public class ContactPage {

    private final WebDriver driver;
    private final WebDriverWait wait;

    private final By nameInput = By.id("na");
    private final By emailInput = By.id("em");
    private final By subjectSelect = By.id("su");
    private final By messageInput = By.id("me");
    private final By fileInput = By.id("fi");

    private final By submitButton =
            By.cssSelector("form#f button.btn.p");

    private final By resultMessage = By.id("m");

    public ContactPage(WebDriver driver) {
        this.driver = driver;

        this.wait = new WebDriverWait(
                driver,
                Duration.ofSeconds(
                        ConfigReader.getInt("explicitWait")
                )
        );
    }

    public ContactPage open() {
        driver.get(ConfigReader.get("baseUrl") + "/contact");

        wait.until(
                ExpectedConditions.visibilityOfElementLocated(nameInput)
        );

        return this;
    }

    private void replaceText(By locator, String value) {
        WebElement element = wait.until(
                ExpectedConditions.visibilityOfElementLocated(locator)
        );

        element.clear();
        element.sendKeys(value);
    }

    public void setName(String value) {
        replaceText(nameInput, value);
    }

    public void setEmail(String value) {
        replaceText(emailInput, value);
    }

    public void setMessage(String value) {
        replaceText(messageInput, value);
    }

    public void selectSubject(String subject) {
        Select select = new Select(
                wait.until(
                        ExpectedConditions.visibilityOfElementLocated(
                                subjectSelect
                        )
                )
        );

        select.selectByVisibleText(subject);
    }

    public void uploadFile(String absolutePath) {
        wait.until(
                ExpectedConditions.presenceOfElementLocated(fileInput)
        ).sendKeys(absolutePath);
    }

    public void submit() {
        wait.until(
                ExpectedConditions.elementToBeClickable(submitButton)
        ).click();
    }

    public void fillValidForm() {
        setName("QA Tester");
        setEmail("qa@example.com");
        selectSubject("Technical problem");
        setMessage(
                "This is a valid support request created by Selenium."
        );
    }

    public String getResultMessage() {
        return wait.until(
                ExpectedConditions.visibilityOfElementLocated(
                        resultMessage
                )
        ).getText();
    }

    public String getMessageValue() {
        return driver.findElement(messageInput)
                .getAttribute("value");
    }

    public String getSelectedSubject() {
        return new Select(
                driver.findElement(subjectSelect)
        ).getFirstSelectedOption().getText();
    }
}

File: playwright/pages/ContactPage.ts
import { expect, Locator, Page } from "@playwright/test";

export class ContactPage {
  readonly page: Page;

  readonly nameInput: Locator;
  readonly emailInput: Locator;
  readonly subjectSelect: Locator;
  readonly messageInput: Locator;
  readonly fileInput: Locator;
  readonly submitButton: Locator;
  readonly resultMessage: Locator;

  constructor(page: Page) {
    this.page = page;

    this.nameInput = page.locator("#na");
    this.emailInput = page.locator("#em");
    this.subjectSelect = page.locator("#su");
    this.messageInput = page.locator("#me");
    this.fileInput = page.locator("#fi");

    this.submitButton = page.locator("form#f button.btn.p");

    this.resultMessage = page.locator("#m");
  }

  async open(): Promise<void> {
    await this.page.goto("/contact");

    await expect(this.nameInput).toBeVisible();
  }

  async setName(value: string): Promise<void> {
    await this.nameInput.fill(value);
  }

  async setEmail(value: string): Promise<void> {
    await this.emailInput.fill(value);
  }

  async selectSubject(subject: string): Promise<void> {
    await this.subjectSelect.selectOption({
      label: subject,
    });
  }

  async setMessage(value: string): Promise<void> {
    await this.messageInput.fill(value);
  }

  async submit(): Promise<void> {
    await this.submitButton.click();
  }

  async fillValidForm(): Promise<void> {
    await this.setName("QA Tester");
    await this.setEmail("qa@example.com");

    await this.selectSubject("Technical problem");

    await this.setMessage("This is a valid Playwright support request.");
  }

  async expectResult(expected: string): Promise<void> {
    await expect(this.resultMessage).toHaveText(expected);
  }
}
